toolkit/tools/pkg/imagecustomizerlib/customizepackages.go

```diff
@@ -15,40 +15,34 @@ import (
 )
 
 func addRemoveAndUpdatePackages(buildDir string, baseConfigPath string, config *imagecustomizerapi.SystemConfig,
-	imageChroot *safechroot.Chroot, rpmsSources []string, useBaseImageRpmRepos bool,
+	imageChroot *safechroot.Chroot, rpmsSources []string, useBaseImageRpmRepos bool, partitionsCustomized bool,
 ) error {
-	allPackagesRemove, err := collectPackagesList(baseConfigPath, config.PackageListsRemove, config.PackagesRemove)
-	if err != nil {
-		return err
-	}
-
-	allPackagesInstall, err := collectPackagesList(baseConfigPath, config.PackageListsInstall, config.PackagesInstall)
-	if err != nil {
-		return err
-	}
-
-	allPackagesUpdate, err := collectPackagesList(baseConfigPath, config.PackageListsUpdate, config.PackagesUpdate)
-	if err != nil {
-		return err
-	}
+	var err error
 
-	needRpmsSources := len(allPackagesInstall) > 0 || len(allPackagesUpdate) > 0 || config.UpdateBaseImagePackages
+	// Note: The 'validatePackageLists' function read the PackageLists files and merged them into the inline package lists.
+	needRpmsSources := len(config.PackagesInstall) > 0 || len(config.PackagesUpdate) > 0 ||
+		config.UpdateBaseImagePackages || partitionsCustomized
 
 	// Mount RPM sources.
 	var mounts *rpmSourcesMounts
 	if needRpmsSources {
-		if len(rpmsSources) <= 0 && !useBaseImageRpmRepos {
-			return fmt.Errorf("have packages to install or update but no RPM sources were specified")
-		}
-
 		mounts, err = mountRpmSources(buildDir, imageChroot, rpmsSources, useBaseImageRpmRepos)
 		if err != nil {
 			return err
 		}
 		defer mounts.close()
 	}
 
-	err = removePackages(allPackagesRemove, imageChroot)
+	if partitionsCustomized {
+		logger.Log.Infof("Updating initrd file")
+
+		err = installOrUpdatePackages("reinstall", []string{"initramfs"}, imageChroot)
```

Comment:
Can we have an option to not use the hostonly dracut flag used here?

Reply:
Nope.

Anytime a package is installed that changes a relevant file (e.g. adding a file under `/etc/dracut.conf.d/), the initrd file will be regenerated using the default config. Ideally, the user would be able to add the `hostonly` option to the dracut config files. But the `mkinitrd` script force sets the `hostonly` value through the CLI. And the regeneration scripts in the `initramfs` package calls `mkinitrd` with default options.

This is a mess and can't be fixed without making breaking changes to Mariner. (It is probably something we should do in Mariner 3.0.)